cvat/apps/events/export.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (C) 2023 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import csv
+from datetime import datetime, timedelta
+from dateutil import parser
+import uuid
+
+import django_rq
+from django.conf import settings
+import clickhouse_connect
+
+
+from rest_framework import serializers, status
+from rest_framework.response import Response
+from django_sendfile import sendfile
+
+from cvat.apps.dataset_manager.views import clear_export_cache, log_exception
+from cvat.apps.engine.log import slogger
+
+DEFAULT_CACHE_TTL = timedelta(hours=1)
+
+def _create_csv(query_params, output_filename, cache_ttl):
+    try:
+        clickhouse_settings = settings.CLICKHOUSE['events']
+
+        time_filter = {
+            'from': query_params.pop('from'),
+            'to': query_params.pop('to'),
+        }
+
+        query = "SELECT * FROM events"
+        conditions = []
+        parameters = {}
+        for param, value in query_params.items():
+            if value:
+                conditions.append(f"{param} = {{{param}:UInt64}}")
+                parameters[param] = value
+
+        if time_filter['from']:
+            conditions.append(f"timestamp >= {{from:DateTime64}}")
+            parameters['from'] = time_filter['from']
+
+        if time_filter['to']:
+            conditions.append(f"timestamp <= {{to:DateTime64}}")
+            parameters['to'] = time_filter['to']
+
+        if conditions:
+            query += " WHERE " + " AND ".join(conditions)
+
+        with clickhouse_connect.get_client(
+            host=clickhouse_settings['HOST'],
+            database=clickhouse_settings['NAME'],
+            port=clickhouse_settings['PORT'],
+            username=clickhouse_settings['USER'],
+            password=clickhouse_settings['PASSWORD'],
+        ) as client:
+            result = client.query(query, parameters=parameters)
+
+        with open(output_filename, 'w', encoding='UTF8') as f:
+            writer = csv.writer(f)
+            writer.writerow(result.column_names)
+            writer.writerows(result.result_rows)
+
+        archive_ctime = os.path.getctime(output_filename)
+        scheduler = django_rq.get_scheduler(settings.CVAT_QUEUES.IMPORT_DATA.value)
+        cleaning_job = scheduler.enqueue_in(time_delta=cache_ttl,
+            func=clear_export_cache,
+            file_path=output_filename,
+            file_ctime=archive_ctime,
+        )
+        slogger.glob.info(
+            f"The {output_filename} is created "
+            f"and available for downloading for the next {cache_ttl}. "
+            f"Export cache cleaning job is enqueued, id '{cleaning_job.id}'"
+        )
+        return output_filename
+    except Exception:
+        log_exception(slogger.glob)
+        raise
+
+def export(request, queue_name):
+    action = request.query_params.get('action', None)
+    filename = request.query_params.get('filename', None)
+
+    query_params = {
```

Comment:
From my perspective, we should implement the functionality using DjangoFilterBackend: https://www.django-rest-framework.org/api-guide/filtering/

Please work with Maxim to understand the best solution here.
Look at the PR: https://github.com/opencv/cvat/pull/5575

Reply:
Discussed with Maxim: DjangoFilterBackend doesn't provide any advantages here due to there is not a django queryset.